internal/util/http_test.go

```diff
@@ -0,0 +1,197 @@
+		req := &http.Request{
+			Header: http.Header{},
+		}
+		versionString := "1.2.3"
+
+		result, _ := addHeader(req, versionString)
+		assert.DeepEqual(t,
+			result.Header[clientHeader],
+			[]string{`{"version":"1.2.3"}`},
+		)
+	})
+}
+
+type MockClient struct {
+	Timeout time.Duration
+}
+
+var funcFoo func() (*http.Response, error)
+
+// Do is the mock request that will return a mock success
+// TODO(benjaminjb): return to this when Operator Version API is complete
+func (m *MockClient) Do(req *http.Request) (*http.Response, error) {
+	return funcFoo()
+}
+
+func TestCheckForUpdates(t *testing.T) {
+	t.Run("success", func(t *testing.T) {
+		// A successful call
+		funcFoo = func() (*http.Response, error) {
+			json := `{"display_name":"PGO 5.0.3","major_version":5}`
+			r := io.NopCloser(strings.NewReader(json))
+			return &http.Response{
+				Body:       r,
+				StatusCode: http.StatusOK,
+			}, nil
+		}
+
+		res, err := checkForUpdates("4.7.3", backoff)
+		assert.NilError(t, err)
+		assert.Equal(t, res, `{"display_name":"PGO 5.0.3","major_version":5}`)
+	})
+
+	t.Run("total failure, err sending", func(t *testing.T) {
+		var counter int
+		// A call returning errors
+		funcFoo = func() (*http.Response, error) {
+			counter++
+			return &http.Response{}, errors.New("whoops")
+		}
+
+		res, err := checkForUpdates("4.7.3", backoff)
+		// Two failed calls because of env var
+		assert.Equal(t, counter, 2)
+		assert.Equal(t, res, "")
+		assert.Equal(t, err.Error(), `whoops`)
+	})
+
+	t.Run("total failure, bad StatusCode", func(t *testing.T) {
+		var counter int
+		// A call returning bad StatusCode
+		funcFoo = func() (*http.Response, error) {
+			counter++
+			return &http.Response{
+				StatusCode: http.StatusBadRequest,
+			}, nil
+		}
+
+		res, err := checkForUpdates("4.7.3", backoff)
+		assert.Equal(t, res, "")
+		// Two failed calls because of env var
+		assert.Equal(t, counter, 2)
+		assert.Equal(t, err.Error(), `received StatusCode 400`)
+	})
+
+	t.Run("one failure, then success", func(t *testing.T) {
+		var counter int
+		// A call returning bad StatusCode the first time
+		// and a successful response the second time
+		funcFoo = func() (*http.Response, error) {
+			if counter < 1 {
+				counter++
+				return &http.Response{
+					StatusCode: http.StatusBadRequest,
+				}, nil
+			}
+			counter++
+			json := `{"display_name":"PGO 5.0.3","major_version":5}`
+			r := io.NopCloser(strings.NewReader(json))
+			return &http.Response{
+				Body:       r,
+				StatusCode: http.StatusOK,
+			}, nil
+		}
+
+		res, err := checkForUpdates("4.7.3", backoff)
+		assert.Equal(t, counter, 2)
+		assert.NilError(t, err)
+		assert.Equal(t, res, `{"display_name":"PGO 5.0.3","major_version":5}`)
+	})
+}
+
+func TestCheckForUpdatesScheduler(t *testing.T) {
+	t.Run("panic recovers", func(t *testing.T) {
+		done := make(chan bool, 1)
+		// capture logs
+		var calls []string
+		logging.SetLogFunc(1, func(input genericr.Entry) {
```

Comment:
🌱 These `SetLogFunc` seem alright for now. I worry about unintended side-effects in other tests; could there be any really? Perhaps this suggests that `CheckForUpdatesScheduler` should take a `context.Context` argument. The purpose of one would be ambiguous, I think: Does the scheduler respect it's `ctx.Done()`? Would canceling it stop the scheduler? etc. Things for some other day.